src/viam/trajex/types/epsilon.hpp

```diff
@@ -77,10 +81,91 @@ class epsilon {
         return epsilon{-value_};
     }
 
+    template <typename T>
+    class wrapper;
+
+    template <typename T>
+    constexpr wrapper<T> wrap(const T& t) const noexcept;
+
    private:
     double value_;
 };
 
+template <typename T>
+class epsilon::wrapper {
+    explicit constexpr wrapper(const epsilon& e, T const& t) : e_{e}, t_{t} {}
+
+    friend class epsilon;
+
+    // clang-format off
```

Comment:
I wonder if we could go to clang-20 in our container and not need to do this anymore. I hate it.